### ISSUE TITLE
Tag exclusion should be detected as invalid write context (same as OR operator)

### ISSUE THREAD

**Issue by bentwitthold** [GothenburgBitFactory/taskwarrior#2664]
Adding a task while active "nowork" context (excluding tasks with the tag "work") prepends "-work" to the description.
Happened after updating to 2.6.0 and with a clean install of 2.6.0 (see below).

* What command(s) did you run?
After auto created .taskrc file:
```
task add +work work task
task add +personal personal task
task context define nowork -work
task context nowork
task add another personal task +personal
task
```
* What did you expect to happen?
There should be a task with the description "another personal task".
```
ID  Age  Tag       Description            Urg
2   54s  personal  personal task          0.8
3   1s   personal  another personal task  0.8
```

* What actually happened?
There is a new task with the description "-work another personal task".
```
ID  Age  Tag       Description                  Urg
2   54s  personal  personal task                0.8
3   1s   personal  -work another personal task  0.8
```
* Paste the output of the ```task diag``` command.
task 2.6.1
   Platform: Linux

Compiler
    Version: Clang 12.0.1
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++17

Build Features
     Commit: 58763fd49
      CMake: 3.21.3
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.7.2
 Build type: release

Configuration
       File: /home/taski/.taskrc (found), 1448 bytes, mode 100644
       Data: /home/taski/.task (found), dir, mode 40755
    Locking: Enabled
         GC: Enabled
     Server: 
         CA: -
Certificate: -
        Key: -
      Trust: strict
    Ciphers: NORMAL
      Creds: 

Hooks
     System: Enabled
   Location: /home/taski/.task/hooks
             (-none-)

Tests
   Terminal: 203x152
       Dups: Scanned 4 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 4 tasks for broken references:
             No broken references found

**Comment by tbabej** [comment]
Hi @bentwitthold, this might be a consequence of the new "write context" feature (check out `task news` after the upgrade!). Can you check how your `context.nowork.write` configuration variable is defined?

**Comment by bentwitthold** [comment]
Yes, `context.nowork.write` was set to `-work`. After removing this it works like expected.

When I define a context with the OR operator, Taskwarrior recognizes the context as not writable.
`task context define workOrPersonal +work or +personal`

When defining a context like this `task context define nowork -work` I would expect that `-work` or any other excluded tag would be omitted from the write context. Or simply give me a similar warning as in the OR context.

I don't know if this can be easily fixed.
Works for me now.

Thanks!

**Comment by tbabej** [comment]
Glad that fixed it!

> Or simply give me a similar warning as in the OR context.

Good suggestion, we should definitely include tag exclusion in the set of cases that emit the warning. We'll fix this in 2.6.2.
